src/xtgeo/xyz/_xyz.py

```diff
@@ -107,36 +94,25 @@ def nrow(self):
 
     def _df_column_rename(self, newname, oldname):
         if isinstance(newname, str):
-            if oldname and self._df is not None:
-                self._df.rename(columns={oldname: newname}, inplace=True)
+            if oldname and self.dataframe is not None:
+                self.dataframe.rename(columns={oldname: newname}, inplace=True)
         else:
             raise ValueError(f"Wrong type of input to {newname}; must be string")
 
     def _check_name(self, value):
         if not isinstance(value, str):
             raise ValueError(f"Wrong type of input; must be string, was {type(value)}")
 
-        if value not in self._df.columns:
+        if value not in self.dataframe.columns:
             raise ValueError(
                 f"{value} does not exist as a column name, must be "
-                f"one of: f{self._df.columns}"
+                f"one of: f{self.dataframe.columns}"
             )
 
+    @abstractmethod
     def copy(self):
-        """Returns a a deep copy of an instance"""
-
-        mycopy = self.__class__()
-        mycopy._df = self._df.copy()
-        mycopy._ispolygons = self._ispolygons
-        mycopy._xname = self._xname
-        mycopy._yname = self._yname
-        mycopy._zname = self._zname
-        mycopy._pname = self._pname
-        mycopy._mname = self._mname
-        mycopy._filescr = self._filesrc = None
-        mycopy._attrs = deepcopy(self._attrs)
-
-        return mycopy
+        """Returns a deep copy of an instance"""
+        ...
 
     def describe(self, flush=True):
```

Comment:
This would be better to have as abstractmethod since the description will be different for Points and Polygons  (but perhaps the intention is to take that in a separate PR?)

Reply:
Yes, lets move that in a separate PR as its the deprecations we need for 2.16.